src/ray/gcs/store_client/redis_store_client.h

```diff
@@ -41,20 +41,101 @@ class RedisStoreClient : public StoreClient {
   Status AsyncGet(const std::string &table_name, const std::string &key,
                   const OptionalItemCallback<std::string> &callback) override;
 
-  Status AsyncGetAll(
-      const std::string &table_name,
-      const SegmentedCallback<std::pair<std::string, std::string>> &callback) override;
+  Status AsyncGetAll(const std::string &table_name,
+                     const ItemCallback<std::unordered_map<std::string, std::string>>
+                         &callback) override;
 
   Status AsyncDelete(const std::string &table_name, const std::string &key,
                      const StatusCallback &callback) override;
 
+  Status AsyncBatchDelete(const std::string &table_name,
+                          const std::vector<std::string> &keys,
+                          const StatusCallback &callback) override;
+
   Status AsyncDeleteByIndex(const std::string &table_name, const std::string &index_key,
                             const StatusCallback &callback) override;
 
  private:
+  /// \class RedisScanner
+  /// This class is used to scan data from Redis.
+  ///
+  /// If you called one method, should never call the other methods.
+  /// Otherwise it will disturb the status of the RedisScanner.
+  class RedisScanner {
+   public:
+    explicit RedisScanner(std::shared_ptr<RedisClient> redis_client,
+                          std::string table_name, std::string match_pattern);
+
+    Status ScanKeysAndValues(
+        const ItemCallback<std::unordered_map<std::string, std::string>> &callback);
+
+    Status ScanKeys(const MultiItemCallback<std::string> &callback);
+
+   private:
+    void Scan(const StatusCallback &callback);
+
+    void OnScanCallback(size_t shard_index, const std::shared_ptr<CallbackReply> &reply,
+                        const StatusCallback &callback);
+
+    void ScanValues(
+        const std::vector<std::string> &keys,
+        const ItemCallback<std::unordered_map<std::string, std::string>> &callback);
+
+    std::string table_name_;
+
+    /// The scan match pattern.
+    std::string match_pattern_;
+
+    /// All keys that received from redis.
+    absl::flat_hash_set<std::string> keys_;
+
+    /// The scan result in rows.
+    /// If the scan type is kScanPartialRows, partial scan result will be saved in this
+    /// variable. If the scan type is kScanAllRows, all scan result will be saved in this
+    /// variable.
+    std::unordered_map<std::string, std::string> rows_;
```

Comment:
update this variable name and comment, as it's not "rows" now.

Reply:
fixed